scripts/check-message-compatibility.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python3
+""" Check message compatibility between two repositories containing a msg/ directory of .msg message definitions """
+
+import os
+import sys
+import difflib
+import re
+import argparse
+
+TOPIC_LIST_FILE = 'px4_ros2_cpp/include/px4_ros2/components/message_compatibility_check.hpp'
+MESSAGES_DEFINE = 'ALL_PX4_ROS2_MESSAGES'
+
+
+def message_fields_str_for_message_hash(topic_type: str, msgs_dir: str) -> str:
+    """
+    Reads the .msg file corresponding to the given topic type, extracts field definitions,
+    and recursively processes nested types to generate a string representation of all fields.
+    """
+    filename = f"{msgs_dir}/msg/{topic_type}.msg"
+    try:
+        with open(filename, 'r') as file:
+            text = file.read()
+    except IOError:
+        print(f"Failed to open {filename}")
+        return ""
+
+    fields_str = ""
+
+    # Regular expression to match field types from .msg definitions
+    msg_field_type_regex = re.compile(
+        r"(?:^|\n)\s*([a-zA-Z0-9_/]+)(\[[^\]]*\])?\s+(\w+)[ \t]*(=)?"
+    )
+
+    # Set of basic types
+    basic_types = {
+        "bool", "byte", "char", "float32", "float64",
+        "int8", "uint8", "int16", "uint16", "int32",
+        "uint32", "int64", "uint64", "string", "wstring"
+    }
+
+    # Iterate over all matches in the text
+    for match in msg_field_type_regex.finditer(text):
+        type_, array, field_name, constant = match.groups()
+
+        if constant == "=":
+            continue
+
+        fields_str += f"{type_}{array} {field_name}\n"
+
+        if type_ not in basic_types:
+            if '/' not in type_:
+                # Recursive call to handle nested types
+                fields_str += message_fields_str_for_message_hash(type_, msgs_dir)
+            else:
+                print(f"Field {filename} contains namespace {type_}")
+
+    return fields_str
+
+
+def hash32_fnv1a_const(s: str) -> int:
+    """Computes the 32-bit FNV-1a hash of a given string"""
+    kVal32Const = 0x811c9dc5
+    kPrime32Const = 0x1000193
+    hash_value = kVal32Const
+    for c in s:
+        hash_value ^= ord(c)
+        hash_value *= kPrime32Const
+        hash_value &= 0xFFFFFFFF
+    return hash_value
+
+
+def message_hash(topic_type: str, msgs_dir: str) -> int:
+    """Generate a hash from a message definition file"""
+    message_fields_str = message_fields_str_for_message_hash(topic_type, msgs_dir)
+    return hash32_fnv1a_const(message_fields_str)
+
+
+def snake_to_pascal(name: str) -> str:
+    """Convert snake_case to PascalCase"""
+    return f'{name.replace("_", " ").title().replace(" ", "")}'
+
+
+def extract_message_type_from_file(filename: str, extract_start_after: str = None,
```

Comment:
I think you're supposed to use `Optional` if it can be None (I did not do it correctly in the first place):
```suggestion
def extract_message_type_from_file(filename: str, extract_start_after: Optional[str] = None,
```